content/articles/line-detection-using-hough-transform-in-matlab/index.md

```diff
@@ -0,0 +1,149 @@
+
+When we execute this code, we get two sinusoids that correspond to the two points in the image, as shown below:
+
+![Two sinusoids corresponding to the two points](/engineering-education/line-detection-using-hough-transform-in-matlab/image-seven.png)
+
+We use the `houghpeaks()` function to find the peaks. Using this function, you can provide the number of peaks required to be returned as the output. The output will be the largest peak if you fail to provide this.
+```Matlab
+hPeaks = houghpeaks(H);
+```
+The output is the largest peak since we did not specify the number of peaks. To see this, execute `hpeaks` in the command window.
+We can find where the peak occurs in the `H` matrix. It is done by indexing theta(T), Rho(R) array, and the pixel location and plotting the output as shown below:
+```Matlab
+x = T(hPeaks(:,2));  %indexing the theta(T) array
+y = R(hPeaks(:,1));  %Indexing the rho(R) array
+
+plot(x,y,'gs')
+```
+The output is:
+
+![Peak](/engineering-education/line-detection-using-hough-transform-in-matlab/image-eight.png)
+
+The green square is the peak location. Peak location is the line that goes through the two image dots.
+Now, we will apply the same logic to the second image. The only difference here is that we extract line segments to redraw the detected lines.
+Lets first plot the line.
+```matlab
+b = eye(150);
+subplot(2,2,3)
+imshow(b)
+title('Image with a line')
+```
+Here, we use the function `eye()` to give an MxN matrix wherewith the ones in the main diagonal and zeros in other parts. When we use this function, we get the plot as shown below:
+
+![plot of line](/engineering-education/line-detection-using-hough-transform-in-matlab/image-nine.png)
+
+In the image, there are many sinusoids. Each of these sinusoids corresponds to a point on a line.
+The last step is to map the peak back into the image plane and extract the line segments. It can be done using `houghlines()` function. This function takes in the original image as the input for the image reference, an array of T, R values and the peaks, which must be marked back into the image plane as the arguments.
+```Matlab
+hlines = houghlines(b, T, R, hPeaks);
+```
+Using these `hline` values, we can redraw the detected lines on the original image. This help us to see if the detection worked. To do this we use the code below:
+```matlab
+xy = [hlines.point1; hlines.point2];
+figure;
+imshow(b)
+hold on
+plot(xy(:,1), xy(:,2), 'g--', 'Linewidth', 5)
+title('Detected line')
+```
+Here, we use the `hlines` structure to create two points, i.e. `hlines.point1` and `hlines.point2`. On top of the original image, we plot the points with different linestyle(--), colour (g) and linewidth(5). It gives the below plot.
+
```

Comment:
Implement the change.

```suggestion

```Matlab
hlines = houghlines(b, T, R, hPeaks);
```

Using these `hline` values, we can redraw the detected lines on the original image. This help us to see if the detection worked. To do this we use the code below:

```matlab
xy = [hlines.point1; hlines.point2];
figure;
imshow(b)
hold on
plot(xy(:,1), xy(:,2), 'g--', 'Linewidth', 5)
title('Detected line')
```

Here, we use the `hlines` structure to create two points, i.e. `hlines.point1` and `hlines.point2`. On top of the original image, we plot the points with different linestyle(--), colour (g) and linewidth(5). It gives the below plot.

```